File: android/Casa/app/src/main/java/fourdognight/github/com/casa/UserVerificationModel.java
package fourdognight.github.com.casa;
import java.util.HashMap;
/**
 * Created by manle on 2/12/2018.
 */

class UserVerificationModel {
    private static HashMap<String, AbstractUser> userList = new HashMap<>();

    static boolean attemptRegistration(String name, String username, String password,
                                       boolean isAdmin) {
        if (userList.containsKey(username)) {
            return false;
        } else if (isAdmin) {
            userList.put(username, new Admin(name, username, password));
        } else {
            userList.put(username, new User(name, username, password));
        }
        return true;
    }

    static AbstractUser attemptLogin(String username, String password) {
        if (!userList.containsKey(username) || !userList.get(username).authenticate(password)) {
            return null;
        } else {
            return userList.get(username);
        }
    }
}

File: ios/iOS App/Casa/UserVerificationModel.swift
//
//  UserVerificationModel.swift
//  Casa
//
//  Created by Isaac Weintraub on 2/19/18.
//  Copyright © 2018 Four Dog Night. All rights reserved.
//

class UserVerificationModel {
    private static var user_list: [String: AbstractUser] = DatabaseInterfacer.getUsers()
    
    static func attemptRegistration(name: String, username: String, password: String, isAdmin: Bool) -> Bool {
        if let _ = user_list[username] {
            return false
        } else if (isAdmin) {
            user_list[username] = Admin(name: name, username: username, password: password)
        } else {
            user_list[username] = User(name: name, username: username, password: password)
        }
        return true
    }
    
    static func attemptLogin(username: String, password: String) -> AbstractUser? {
        let user: AbstractUser? = user_list[username]
        if (user == nil || !user!.authenticate(password: password)) {
            return nil
        }
        return user
    }
}
